llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -3362,10 +3385,23 @@ Instruction *InstCombinerImpl::visitAllocSite(Instruction &MI) {
     DIB.reset(new DIBuilder(*MI.getModule(), /*AllowUnresolved=*/false));
   }
 
-  if (isAllocSiteRemovable(&MI, Users, TLI)) {
+  // Determine what getInitialValueOfAllocation would return without actually allocating the result.
+  bool KnowInitUndef = isa<AllocaInst>(MI);
```

Comment:
Don't need the special case for AllocaInst, getInitialValueOfAllocation will handle it?

Reply:
Seems like a good idea, but it doesn't look like it: https://llvm.org/doxygen/namespacellvm.html#a108b6e33f153eda5019d322f0ac909b0